bugsnag-android-core/src/main/java/com/bugsnag/android/internal/ImmutableConfig.kt

```diff
@@ -181,6 +182,34 @@ internal fun convertToImmutableConfig(
     )
 }
 
+private fun validateApiKey(value: String, logger: Logger?) {
+    if (isInvalidApiKey(value)) {
+        logger?.w(
+            "Invalid configuration. " +
+                "apiKey should be a 32-character hexademical string, got " + value
```

Comment:
I think we should be using `DebugLogger` to avoid hiding the log message when a customer or no-op `Logger` is being used.

```suggestion
private fun validateApiKey(value: String) {
    if (isInvalidApiKey(value)) {
        DebugLogger.w(
            "Invalid configuration. apiKey should be a 32-character hexademical string, got $value"
```